games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+
+      await this.delay(2000);
+      // eslint-disable-next-line no-else-return
+    } else {
+      // house
+      const randomIndex = Math.floor(
+        Math.random() * this.players[1].hand.length
+      );
+      const discardedCard =
+        this.players[1].hand[randomIndex];
+      if (discardedCard) {
+        this.children.bringToTop(discardedCard);
+        discardedCard.playMoveTween(
+          this.deckFrontCardX,
+          this.deckFrontCardY
+        );
+        discardedCard.playFlipOverTween();
+        this.deckFront.push(discardedCard);
+        player.removeCardFromHand(discardedCard);
+      }
+    }
+  }
+
+  private playerHandSelectedCard(): Promise<void> {
+    return new Promise<void>((resolve) => {
+      this.input.setDraggable(this.players[0].hand);
+      this.createDropZoneEvent().then(() => {
+        this.input.setDraggable(
+          this.players[0].hand,
+          false
+        );
+        resolve();
+      });
+    });
+  }
+
+  private checkIfCardOverDeck(card: Card) {
+    const deckBounds = this.deckFront[0].getBounds();
+    const cardBounds = card.getBounds();
+
+    return Phaser.Geom.Intersects.RectangleToRectangle(
+      deckBounds,
+      cardBounds
+    );
+  }
+
+  // メルドゾーンにおくカードを選定
+  // eslint-disable-next-line class-methods-use-this
+  private findMeldAbleCard(hands: Card[]) {
+    const meldAbleCards: Card[][][] = [[], []];
+
+    // 同じrankが3枚以上ある場合の処理
+    const rankCounts: Record<string, number> = {};
+    hands.forEach((card) => {
+      if (rankCounts[card.rank]) rankCounts[card.rank] += 1;
+      else rankCounts[card.rank] = 1;
+    });
+
+    Object.keys(rankCounts).forEach((rank) => {
+      if (rankCounts[rank] >= 3) {
+        const cards = hands.filter(
+          (card) => card.rank.toString() === rank.toString()
+        );
+        meldAbleCards[0].push([...cards]);
+      }
+    });
+
+    // 同じsuitで連続するrankの組み合わせを探す
+    const suits: string[] = Array.from(
+      new Set(hands.map((card) => card.suit))
+    );
+    suits.forEach((suit) => {
+      const sortedHand = hands
+        .filter(
+          (card) => card.suit.toString() === suit.toString()
+        )
+        .sort(
+          (a, b) =>
+            a.getRankNumber('rummy') -
+            b.getRankNumber('rummy')
+        );
+
+      for (let i = 0; i < sortedHand.length - 2; i += 1) {
+        const currRanks = sortedHand.slice(i, i + 3);
+        const isConsecutive = currRanks.every(
+          (card, index) =>
+            card.getRankNumber('rummy') ===
+            currRanks[0].getRankNumber('rummy') + index
+        );
+
+        if (isConsecutive) {
+          meldAbleCards[1].push([...currRanks]);
+        }
+      }
+    });
+
+    console.log(meldAbleCards[0]);
+    console.log(meldAbleCards[1]);
+
+    return meldAbleCards.length > 0
+      ? meldAbleCards
+      : [[], []];
+  }
+
+  // meldZoneへ移動させる
+  private async cardPutOnMeldZone(player: RummyPlayer) {
```

Comment:
- 関数が大きすぎるので、中の処理を関数として切り出してください。
- 関数名が意味分からないです。メルドゾーンにカードを出す関数と理解したので、putDownCardToMeldZoneなどはどうでしょうか？


```suggestion
  private async putDownCardToMeldZone(player: RummyPlayer) {
```